packages/core/src/agent/MessageSender.ts

```diff
@@ -93,26 +115,31 @@ export class MessageSender {
     }
 
     // Retrieve DIDComm services
-    const allServices = this.transportService.findDidCommServices(connection)
+    const allServices = this.transportService.findDidCommServices(connection, this.supportedTransports)
```

Comment:
As I understand the code now, it seems like `didcomm:transport/queue` is not taken into account here. Maybe I'm missing something, but that would mean all `didcomm:tranport/queue` services are ignored.

As I'm thinking of it right now, maybe we can do something like the following:

1. `findDidCommServices` returns a diddoc based priority sorted list of all services (already did that)
2. We loop through the didoc services and for each service we check if we can send it using one of the registered transports (`transport.suportedSchemes`)
3. If none of the inbound or outbound transports can deliver the message, we check if a queue service is included. If this is the case, we queue the message


Addition could be made between 1. and 2. where we sort the return services from `findDidCommServices`, based on a preferred protocol scheme for that message. This would be useful in the mediation recipient class where we do message pickup from the mediator.

Reply:
if I am not mistaken, sending a message on a connection with their did_doc containing only services with `didcomm:transport/queue` would error without explicit details, no queueing supported. The code ignores them. This will need to be updated with the check for a queue service described in suggestion 3.

Moving the preferred protocol scheme sort into the mediation recipient class would work for message pickup from mediators, but would not catch any other message sending. A nice feature of having the sort in the message sender is speed. While establishing a connection between an edge agent and an agent that supports WebSockets, the edge agent may initialize the connection protocol using HTTP, once the edge agent receives the other agent's did_doc the edge agent will finish establishing the connection using WebSockets. Why have multiple services in a did_doc if we are restricted to use the service a connection invitation is created with?

We could try and build a preferred protocol scheme by using `findDidCommServices` on senders did_doc. This will return the sender's inbound services, not the senders outbound services. I assumed we would need to update the did_doc services to have services that define sending services alongside the receiving services. To avoid changing/learning the definition of a service in a did_doc and supported code changes we opted to do list sorting and filtering based on a list of supported schemas returned from a list of registered transporters.
If you could provide an example of a did_doc with services that represent supported outbound transports we can update the message sender to use this. Another issue arises if we use the sender's did_doc services to build a preferred protocol list. connections and the did_docs associated are persisted, while transporters are not. We would need to check the did_doc services at the start of AFJ and update them along with the agent on the other end of the connection.

Reply:
I have updated the code. `didcomm:transport/queue` is being handled correctly, I didn't notice that after the merge, no need to fix it. 
 The proposed flow is the intended behavior of this code except for priority sort. instead of moving priority sorting into mediation messages I have added an optional parameter priority with a default in send message and send package. `sendMessage(outboundMessage: OutboundMessage, priority = false)`. 
 If priority is true we pass in a secondary list to findDidCommServices, otherwise an empty list. 
 ```
     // Retrieve DIDComm services
    let allServices: Array<DidCommService | IndyAgentService>
    if (priority) {
      allServices = this.transportService.findDidCommServices(connection, this.supportedTransportSchemes)
    } else {
      allServices = this.transportService.findDidCommServices(connection, [])
    }
 ```
 Now any messaging were a consumer of AFJ wishes to prioritize supported transports can be sent by setting the priority to true. Enabling desired features for mediated messages while keeping the current functionality of the message sender.